### PR TITLE
various cleanups, removal of warnings and redundancies, fixes of code formattings

### PR DESCRIPTION
**Pull request by pitr**
various cleanups, removal of warnings and redundancies, fixes of code formattings

### CODE REVIEW

**Comment by pitr** [comment]
review with whitespace hidden (`?w=1`)

**Review by alex-on-java — COMMENTED** [review]
Thanks a lot for taking care of the code quality and readability. And I'm even more grateful that you did it in a separate PR so that I can easily review it.

**Comment by alex-on-java** [comment]
👍

File: src/main/java/org/kairosdb/core/datastore/CachedSearchResult.java [hunk 1]
```diff
@@ -29,471 +29,400 @@
...
+        }
+    }
+
+    /**
+     * Reads the index file into memory
+     */
+    private void loadIndex() throws IOException, ClassNotFoundException {
+        ObjectInputStream in = new ObjectInputStream(new FileInputStream(m_indexFile));
+        int size = in.readInt();
+        for (int I = 0; I < size; I++) {
```
[the remark this answers is not in the document]
Comment (pitr) [reply src/main/java/org/kairosdb/core/datastore/CachedSearchResult.java]: fixed

**Comment by pitr** [comment]
👍

**Comment by alex-on-java** [comment]
👍

**Pull request merged**